azext_iot/tests/iothub/devices/test_hub_export_import.py

```diff
@@ -0,0 +1,364 @@
+            ).get_output_in_json()
+            target_module_twin = self.cmd(
+                f"iot hub module-twin show -m {module['moduleId']} -d {device1['deviceId']} -l {self.dest_hub_cstring}"
+            ).get_output_in_json()
+
+            assert(len(module_twin["properties"]["desired"]) == len(target_module_twin["properties"]["desired"]))
+            for prop in module_twin["properties"]["desired"]:
+                if(prop != "$metadata" and prop != "version"):
+                    assert(prop in target_module_twin["properties"]["desired"])
+                    assert(module_twin["properties"]["desired"][prop] == target_module_twin["properties"]["desired"][prop])
+
+            if("tags" in module_twin.keys()):
+                assert(module_twin["tags"] == target_module_twin["tags"])
+            assert(module_twin["status"] == target_module_twin["status"])
+
+    def compare_hubs(self):
+
+        # compare configurations (there's only one)
+
+        orig_hub_configs = self.cmd(
+            f"iot hub configuration list -l {self.connection_string}"
+        ).get_output_in_json()
+
+        dest_hub_configs = self.cmd(
+            f"iot hub configuration list -l {self.dest_hub_cstring}"
+        ).get_output_in_json()
+
+        assert(len(orig_hub_configs) == len(dest_hub_configs) == 1)
+
+        self.compare_configs(orig_hub_configs[0], dest_hub_configs[0])
+
+        # compare devices
+
+        orig_hub_identities = self.cmd(
+            f"iot hub device-identity list -l {self.connection_string}"
+        ).get_output_in_json()
+        dest_hub_identities = self.cmd(
+            f"iot hub device-identity list -l {self.dest_hub_cstring}"
+        ).get_output_in_json()
+
+        for device in orig_hub_identities:
+            target = None
+            for id in dest_hub_identities:
+                if(device["deviceId"] == id["deviceId"]):
+                    target = id
+                    break
+
+            assert target
+
+            self.compare_devices(device, target)
+
+    def clean_up_dest_hub(self):
+
+        dest_hub_configs = self.cmd(
+            f"iot hub configuration list -l {self.dest_hub_cstring}"
+        ).get_output_in_json()
+
+        for config in dest_hub_configs:
+            self.cmd(
+                "iot hub configuration delete -c {} -l {}".format(config["id"], self.dest_hub_cstring)
+            )
+
+        dest_hub_identities = self.cmd(
+            f"iot hub device-identity list -l {self.dest_hub_cstring}"
+        ).get_output_in_json()
+
+        for id in dest_hub_identities:
+            self.cmd(
+                "iot hub device-identity delete -d {} -l {}".format(id["deviceId"], self.dest_hub_cstring)
+            )
+
+        # gives the api enough time to update
+        time.sleep(1)
+
+    def test_import_export(self):
+
+        # auth-type = key (default)
```

Comment:
you can use the `for auth_phase in DATAPLANE_AUTH_TYPES:` and `self.set_cmd_auth_type` to do the whole auth type mess for you